extensions/github/src/pushErrorHandler.ts

```diff
@@ -128,6 +140,60 @@ async function handlePushError(repository: Repository, remote: Remote, refspec:
 	})();
 }
 
+async function getPullRequestTemplates(repository: Repository): Promise<Uri[]> {
+	/**
+	 * Places a PR template can be:
+	 * - At the root, the docs folder, or the.github folder, named pull_request_template.md or PULL_REQUEST_TEMPLATE.md
+	 * - At the same folder locations under a PULL_REQUEST_TEMPLATE folder with any name
+	 *
+	 * NOTE This method is a modified copy of a method with same name at microsoft/vscode-pull-request-github repository:
+	 *   https://github.com/microsoft/vscode-pull-request-github/blob/0a0c3c6c21c0b9c2f4d5ffbc3f8c6a825472e9e6/src/github/folderRepositoryManager.ts#L1061
+	 *
+	 */
+	const pattern1 = '{pull_request_template,PULL_REQUEST_TEMPLATE}.md';
+	const templatesPattern1 = await workspace.findFiles(
+		new RelativePattern(repository.rootUri, pattern1)
+	);
+
+	const pattern2 = '{docs,.github}/{pull_request_template,PULL_REQUEST_TEMPLATE}.md';
+	const templatesPattern2 = await workspace.findFiles(
+		new RelativePattern(repository.rootUri, pattern2), null
+	);
+
+	const pattern3 = 'PULL_REQUEST_TEMPLATE/*.md';
+	const templatesPattern3 = await workspace.findFiles(
+		new RelativePattern(repository.rootUri, pattern3)
+	);
+
+	const pattern4 = '{docs,.github}/PULL_REQUEST_TEMPLATE/*.md';
+	const templatesPattern4 = await workspace.findFiles(
+		new RelativePattern(repository.rootUri, pattern4), null
+	);
```

Comment:
This might actually get quite expensive. The first two can be replaced by simple `stat` calls. The second two could also be replaced with `readdir` calls. Could we do that instead?

Also, we could potentially do it all in parallel, instead of in sequence as is right now.

Reply:
I need to mention I took these, mostly intact, from [`microsoft/vscode-pull-request-github`][source] to ensure thorough Github template location.

I agree with you and will make the changes.

[source]: https://github.com/microsoft/vscode-pull-request-github/blob/0a0c3c6c21c0b9c2f4d5ffbc3f8c6a825472e9e6/src/github/folderRepositoryManager.ts#L1061

Reply:
cc @alexr00

Reply:
They can get expensive, but `workspace.findFiles` is browser friendly, which is needed for microsoft/vscode-pull-request-github.

Reply:
@alexr00 Still we could do things in parallel for better performance, right?

Reply:
Certainly, just pushed the fix for that.